lms/djangoapps/branding/views.py

```diff
@@ -102,3 +114,223 @@ def courses(request):
+
+
+def _render_footer_html(show_openedx_logo):
+    """Render the footer as HTML.
+
+    Arguments:
+        show_openedx_logo (bool): If True, include the OpenEdX logo in the rendered HTML.
+
+    Returns: unicode
+
+    """
+    context = {'hide_openedx_link': not show_openedx_logo}
+    return (
+        render_to_response("footer-edx-v3.html", context)
+        if settings.FEATURES.get("IS_EDX_DOMAIN", False)
+        else render_to_response("footer.html", context)
+    )
+
+
+def _footer_css_name():
+    """Return the name of the footer CSS static file.
+
+    This takes into account:
+    1) Whether asset pipelining is being used (handles both prod and local dev)
+    2) Text direction (right-to-left support)
+
+    Returns: unicode
+
+    """
+    bidi = 'rtl' if translation.get_language_bidi() else 'ltr'
+    version = 'edx' if settings.FEATURES.get('IS_EDX_DOMAIN') else 'openedx'
+    css_name = settings.FOOTER_CSS[version][bidi]
+
+    # In production, the asset pipeline copies compiled sass
+    # to a css folder.  Unfortunately, this doesn't happen
+    # in devstack, so we load the version of the CSS that
+    # hasn't yet gone through the asset pipeline.
+    return (
+        u"sass/{name}" if settings.DEBUG
+        else u"css/{name}"
+    ).format(name=css_name)
+
+
+def _send_footer_static(request, name):
+    """Send static files for the footer.
+
+    Send static files required for rendering the footer
+    client-side (JavaScript and CSS).
+
+    Ordinarily, clients would load static resources like
+    this from /static/ URLs.  In production, these URLs always
+    include cache-busting hashes.  Since these URLs are meant
+    to be included directly in the DOM, we want a URL
+    that is always available (e.g. /api/v1/branding/footer.css).
+
+    On the other hand, we *don't* want to serve these
+    files directy through Django, because nginx is much
+    faster at this.
+
+    For this reason, we use `django-sendfile` as a compromise.
+    In production, this will use the "X-Accel-Redirect" header
+    to trigger an internal redirect to the static file.
+    That way, Django can resolve the path, but nginx still
+    serves the file.
+
+    If the file can't be found, this will return a 404.
+
+    Arguments:
+        request (HttpRequest)
+        name (unicode): The name of the static file to load
+            (relative to `settings.STATIC_ROOT`)
+
+    Returns:
+        HttpResponse
+    """
+    # In production, we will load assets from the STATIC_ROOT
+    # directory.  In local development, we don't run collectstatic,
+    # so we need to find the files in their original locations.
+    static_path = (
+        staticfiles_find(name) if settings.DEBUG
+        else staticfiles_storage.path(name)
+    )
+    if static_path is None:
+        raise Http404
+
+    return sendfile(request, static_path)
+
+
+def footer(request, extension="json"):
+    """Retrieve the branded footer.
```

Comment:
@mhoeber This docstring is what I'd imagine we'd want in the documentation for this API end-point.